TS-4211: Make freelist_new cleaner so we can debug better.

PSUdaemon opened the pull request.

bgaff commented on lib/ts/ink_queue.cc:
```diff
@@ -200,19 +200,20 @@ freelist_new(InkFreeList *f)
       void *newp = NULL;
-      size_t alloc_size = 0;
+      size_t alloc_size = f->chunk_size * f->type_size;
+      size_t alignment;
```
= 0.

bgaff commented:
All good :+1: just set alignment = 0 on declaration and let the complier deal with it if it's always assigned later.

bgaff commented:
I just confirmed that this is correct, it will actually be hugepage aligned.

PSUdaemon commented:
I addressed the comments above. At some point in the future I want to replace this with a page allocator for the freelists but that is going to be a bigger task.

PSUdaemon commented:
Merged.
